### PR TITLE
Errcheck linter

### PR DESCRIPTION
**Pull request by vishwas-sharma2480**
Summary of Changes

Enabled the errcheck linter rule.
Reviewed and fixed all linter errors related to the errcheck rule.
Updated the codebase to ensure that error return values are checked when necessary.
Removed the line that disables the errcheck rule, enabling it for the project.
Additional Information

Closes: #2346 
Enabled the errcheck rule to improve code quality and error handling.

### CODE REVIEW

**Comment by BupycHuk** [comment]
Hello, could you please fix the merge conflicts?

File: .golangci.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ linters-settings:
 
   godot:
     capital: true
-    period: false
+    period: true
```
Comment (atymchuk) [inline .golangci.yml:32]: Not sure if your branch is based on main, but this value is already set to rue in `main`.

https://github.com/percona/pmm/blob/6e8cc78054950739598adb182ef16d6f851f35e5/.golangci.yml#L32

Comment (atymchuk) [reply .golangci.yml:32]: Moreover, you previous PR was merged, so there is no need to put the dots at the end of comments. Please double check to make sure your branch is based on the latest `main`.

**Comment by vishwas-sharma2480** [comment]
Hi Alex
Look like my branch was made from the branch which did not have these
changes i will try and fix this

**Pull request closed without merging**